src/besseli.jl

```diff
@@ -1,46 +1,99 @@
+# Modified Bessel functions of the first kind of order zero and one
+# besseli0, besseli1
+# Scaled modified Bessel functions of the first kind of order zero and one
+# besseli0x, besselix
+
 #=
-Cephes Math Library Release 2.8:  June, 2000
-Copyright 1984, 1987, 2000 by Stephen L. Moshier
-https://github.com/jeremybarnes/cephes/blob/master/bessel/i0.c
-https://github.com/jeremybarnes/cephes/blob/master/bessel/i1.c
+Approximate forms used here are given in
+"Rational Approximations for the Modified Bessel Function of the First Kind - I1(x) for Computations with Double Precision"
+by Pavel Holoborodko, 
+http://www.advanpix.com/2015/11/12/rational-approximations-for-the-modified-bessel-function-of-the-first-kind-i1-for-computations-with-double-precision/
+Actual coefficients used are from the boost math library.
+https://github.com/boostorg/math/tree/develop/include/boost/math/special_functions/detail
 =#
 function besseli0(x::T) where T <: Union{Float32, Float64}
-    if x <= 8
-        y = muladd(x, T(.5), T(-2))
-        return exp(x) * chbevl(y, A_i0(T))
+    x = abs(x)
+    if x < 7.75
+        a = x * x / 4
+        return muladd(a, evalpoly(a, besseli0_small_coefs(T)), 1)
     else
-        return exp(x) * chbevl(T(32) / x - T(2), B_i0(T)) / sqrt(x)
+        a = exp(x / 2)
+        s = a * evalpoly(inv(x), besseli0_med_coefs(T)) / sqrt(x)
+        return a * s
     end
 end
 function besseli0x(x::T) where T <: Union{Float32, Float64}
+    T == Float32 ? branch = 50 : branch = 500
     x = abs(x)
-    if x <= 8
-        y = muladd(x, T(.5), T(-2))
-        return chbevl(y, A_i0(T))
+    if x < 7.75
+        a = x * x / 4
+        return muladd(a, evalpoly(a, besseli0_small_coefs(T)), 1) * exp(-x)
+    elseif x < branch
+        return evalpoly(inv(x), besseli0_med_coefs(T)) / sqrt(x)
+    else
+        return evalpoly(inv(x), besseli0_large_coefs(T)) / sqrt(x)
+    end
+end
+function besseli1(x::Float32)
```

Comment:
This `Float32` implimentation is exactly the same as the `Float64` version, should they be merged?

Reply:
Yes good catch! Removing the branch in the Float64 makes it equal to the Float32 now. I have merged them. I also went and merged the scaled functions to match.